packages/node_modules/@ciscospark/redux-module-avatar/src/actions.js

```diff
@@ -57,40 +52,55 @@ export function fetchAvatar({space, userId}, spark) {
       return Promise.resolve();
     }
 
-    dispatch(addAvatarBegin(avatarId));
-
     if (space) {
-      return fetchSpaceAvatar(space, spark, userId)
-        .then(({id, objectUrl}) => {
-          dispatch(addAvatar(id, objectUrl));
-        });
+      return dispatch(fetchSpaceAvatar(space, spark, userId));
     }
-    return fetchUserAvatar(userId, spark)
-      .then(({id, objectUrl}) => {
-        dispatch(addAvatar(id, objectUrl));
-      });
+    return dispatch(fetchUserAvatar(userId, spark));
   };
 }
 
+
 function fetchUserAvatar(userId, spark) {
-  return spark.people.get(userId)
-    .then((person) => Promise.resolve({id: userId, objectUrl: person.avatar}))
-    .catch(() => Promise.resolve({id: userId}));
+  return (dispatch) => {
+    dispatch(addAvatarBegin(userId));
+    return spark.people.get(userId)
+      .then((person) => dispatch(addAvatar(userId, convertToSmallAvatar(person.avatar))))
+      .catch(() => {
+        dispatch(addAvatar(userId));
+        return Promise.resolve({id: userId});
+      });
+  };
 }
 
-function fetchSpaceAvatar(space, spark, userIdForAvatar) {
-  if (space.type === `direct`) {
-    if (!userIdForAvatar) {
-      return Promise.reject(new Error(`Direct spaces require a user id to display`));
+function fetchSpaceAvatar(space, sparkInstance, userIdForAvatar) {
+  return (dispatch) => {
+    if (space.type === `direct` && space.participants.length === 2) {
```

Comment:
Did you find a condition where a direct space had something other than 2 participants?

Reply:
Yea, there were direct spaces that only had 1 participant, the logged in user. When that happened it would throw.

Reply:
Ah yea, that's right, when a user gets deleted